code/modules/reagents/reagent_containers/cups/glassbottle.dm

```diff
@@ -39,11 +39,93 @@
 	var/bottle_knockdown_duration = BOTTLE_KNOCKDOWN_DEFAULT_DURATION
 	tool_behaviour = TOOL_ROLLINGPIN // Used to knock out the Chef.
 	toolspeed = 1.3 //it's a little awkward to use, but it's a cylinder alright.
+	/// A contained piece of paper, a photo, or space cash, that we can use as a message or gift to future spessmen.
+	var/obj/item/message_in_a_bottle
 
 /obj/item/reagent_containers/cup/glass/bottle/Initialize(mapload, vol)
 	. = ..()
 	var/static/list/recipes =  list(/datum/crafting_recipe/molotov)
 	AddElement(/datum/element/slapcrafting, recipes)
+	register_context()
+	register_item_context()
+
+/obj/item/reagent_containers/cup/glass/bottle/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	if(message_in_a_bottle)
+		return NONE
+	if(istype(held_item, /obj/item/paper) || istype(held_item, /obj/item/stack/spacecash) || istype(held_item, /obj/item/photo))
+		context[SCREENTIP_CONTEXT_LMB] = "Insert message"
+		return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
+/obj/item/reagent_containers/cup/glass/bottle/add_item_context(obj/item/source, list/context, atom/target, mob/living/user)
+	if(message_in_a_bottle && HAS_TRAIT(target, TRAIT_MESSAGE_IN_A_BOTTLE_LOCATION))
+		context[SCREENTIP_CONTEXT_RMB] = "Toss message"
+		return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
+/obj/item/reagent_containers/cup/glass/bottle/Exited(atom/movable/gone, atom/newloc)
+	if(gone == message_in_a_bottle)
+		message_in_a_bottle = null
+		if(!QDELETED(src))
+			update_icon(UPDATE_OVERLAYS)
+	return ..()
+
+/obj/item/reagent_containers/cup/glass/bottle/CheckParts(list/parts_list)
+	. = ..()
+	var/obj/item/reagent_containers/cup/glass/bottle/bottle = locate() in contents
+	if(bottle.message_in_a_bottle)
+		message_in_a_bottle = bottle.message_in_a_bottle
+		bottle.message_in_a_bottle.forceMove(src)
+
+/obj/item/reagent_containers/cup/glass/bottle/examine(mob/user)
+	. = ..()
+	if(message_in_a_bottle)
+		. += span_info("there's \a [message_in_a_bottle] inside it. Break it to take it out, or find a beach or ocean and toss it with [EXAMINE_HINT("right-click")].")
+	else if(isGlass)
+		. += span_tinynoticeital("you could place a paper, photo or space cash inside it...")
+
+/obj/item/reagent_containers/cup/glass/bottle/update_overlays()
+	if(message_in_a_bottle)
+		var/overlay = add_message_overlay()
+		if(overlay)
+			. += overlay
+	return ..()
```

Comment:
adding the overlay to the default return does nothing because you direct return ..() at the end

Reply:
Whoops.